torch/_inductor/ir.py

```diff
@@ -3912,12 +4011,39 @@ def codegen_list_tuple_access(self, basename, indices):
         else:
             return basename
 
+    def get_unbacked_symbol_defs(self):
+        # This kernel defines all unbacked symbols... that it didn't get in as
+        # arguments!
+        defs = (
+            free_unbacked_symbols(self.get_size())
+            | free_unbacked_symbols(self.get_stride())
+            | free_unbacked_symbols(self.get_offset())
+        )
+        return defs - self.get_unbacked_symbol_uses()
+
     def codegen(self, wrapper):
         wrapper.codegen_multi_output(
             self.get_name(),
             self.codegen_list_tuple_access(self.inputs[0].get_name(), self.indices),
         )
-        self.codegen_size_asserts(wrapper)
+        # NB: If it is possible for other ir node types to return unbacked
+        # symints, we also have to add this logic there too.  Don't forget to
+        # update get_unbacked_symbol_defs too.
+        symbols_to_define = self.get_unbacked_symbol_defs()
+        for i, s in enumerate(self.get_size()):
+            if s in symbols_to_define:
+                wrapper.writeline(f"{s} = {self.get_name()}.size({i})")
+                symbols_to_define.remove(s)
+        for i, s in enumerate(self.get_stride()):
+            if s in symbols_to_define:
+                wrapper.writeline(f"{s} = {self.get_name()}.stride({i})")
+                symbols_to_define.remove(s)
+        if (s := self.get_offset()) in symbols_to_define:
+            wrapper.writeline(f"{s} = {self.get_name()}.storage_offset()")
```

Comment:
Can we use `wrapper.declare` and `wrapper.ending` so that this also works in C++ wrapper codegen? Smth like this:

```
wrapper.writeline(f"{wrapper.declare}{s} = {self.get_name()}.size({i}){wrapper.ending}")
```

Reply:
i'm going to do this in a follow up, since there are some PRs backed up on this landing